src/main/java/edu/umich/ctools/sectionsUtilityTool/Utils.java

```diff
@@ -131,4 +132,43 @@ public static FileReader retrieveTestFile(String url, FileReader fr, String msg,
 		}
 		return fr;
 	}
+
+	public static void logApiCall(HttpServletRequest request) {
+		String uniqname = null;
+		String queryString = request.getQueryString();
+		String pathInfo = request.getPathInfo();
+		String url = pathInfo;
+		if (queryString != null) {
+			url = url + "?" + queryString;
+		}
+
+		TcSessionData tc = (TcSessionData) request.getSession().getAttribute(TC_SESSION_DATA);
+		M_log.debug("TC Session Data: " + tc);
+
+		String baseString = "FRIEND API request with Uniqname \"%s\" for URL \"%s\"";
+
+		if (tc != null) {
+			uniqname = (String) tc.getCustomValuesMap().get("custom_canvas_user_login_id");
+		}
+		logMsg(uniqname, url, request, baseString);
+	}
+
+	public static void logApiCall(String uniqname, String originalUrl, HttpServletRequest request) {
+		String baseString = "CANVAS API request with Uniqname \"%s\" for URL \"%s\"";
+		logMsg(uniqname, originalUrl, request, baseString);
+	}
+
+	private static void logMsg(String uniqname, String originalUrl, HttpServletRequest request, String baseString) {
+		String remoteUser = request.getRemoteUser();
+
+		String loggingApiWithSessionInfo;
+		if (uniqname != null) {
```

Comment:
Since the real choice in this if is the name to be used for the log message it would be clearer to have the if just set a variable with the chosen name and have just one call to String.format.

Also, to my surprise, log has simple formatting built in.  It will replace brackets "{}" in a message with corresponding variables provided after the message string.  I'm sure it doesn't work for everything but it often makes writing log messages much quicker.